### ISSUE TITLE
KVM: corrupted source files

### ISSUE THREAD

**Issue by bmwiedemann** [openSUSE/obs-build#1030]
For my work on reproducible builds I am using `osc build --vm-type=kvm` a lot and noticed that the VM occasionally gets corrupted build input files on a on Leap 15.5 system.

For .spec .changes and .rpm (BuildRequires) the build fails early, but now that I started using pbuild I also found 3 cases where the .src.rpm silently contained the corrupted file (cpanspec.yml, rpmlintrc etc).
The corrupted file had the same size but contained binary garbage (and `build-compare` would not even report a diff).

The 2nd stage should verify all checksums of build inputs before starting to build, so silent corruption gets avoided and a consistent build error message is produced.

This was with `build-20240313-150500.446.1` from openSUSE:Tools but has been going on for a longer time.

**Comment by bmwiedemann** [comment]
I am now trying `osc build --build-opt=--vm-use-mkfs-copyin` as a possible workaround. Should have results next week.

**Comment by adrianschroeter** [comment]
so you are building as root? (otherwise mkfs-copyin would be default).

If there are corupted files without that mode I would verify if the umount did not happen for some reason before starting the VM.  (As I can not really believe that the kernel is corrupting the block cache, but that would be another option).

**Comment by bmwiedemann** [comment]
Yes, that was building as root. I did not see this corruption with mkfs-copyin so far.
